API/Controllers/SearchController.cs

```diff
@@ -0,0 +1,65 @@
+using API.DTOs;
+using API.Services.Interfaces;
+using Asp.Versioning;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+[ApiController]
+[ApiVersion(1)]
+[EnableCors]
+[Authorize(Roles = "system")] // Internal access only at this time
+[AllowAnonymous]
+[Route("api/v{version:apiVersion}/[controller]")]
+public class SearchController(ITournamentsService tournamentsService, IMatchesService matchesService, IPlayerService playerService) : Controller
+{
+    [HttpGet]
+    public async Task<IActionResult> Test()
+    {
+        return Ok();
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> SearchByNames([FromQuery(Name = "tournamentName")] string? tournamentName, [FromQuery(Name = "matchName")] string? matchName, [FromQuery(Name = "username")] string? username)
```

Comment:
The `Name = ""` is not necessary on `FromQuery` if it's identical to the method param.

Reply:
Fixed in latest commit